### DISCUSSION TITLE
Modify message deadline/timeout by the natsgo client

### DISCUSSION THREAD

**Discussion by in-cloud-opensource** [nats-io/nats.go#962] [Q&A] [answered]
Is there a way to set the acknowledge deadline for a single message by the nats client?

**Comment by derekcollison** [comment, marked as answer]
This is set per consumer. However if you need more time for an individual message you can call m.Progress() which resets the timer. So if you have AckWait of 30s, and you know you need to take longer, calling ack.Progress() will reset the time to another 30s from when Progress() is called.
